### ISSUE TITLE
java.util.MissingResourceException: Can't find bundle for base name org.apache.tomcat.util.descriptor.tld.LocalStrings, locale en_US

### ISSUE THREAD

**Issue by frankye2099** [spring-projects-experimental/spring-graalvm-native#332]
got error when run the native build.

`
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.4.0-M4</version>
		<relativePath/> 
	</parent>
	<groupId>com.example</groupId>
	<artifactId>rest-service</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<name>rest-service</name>
	<description>Demo project for Spring Boot</description>

	<properties>
		<java.version>1.8</java.version>
	</properties>

	<dependencies>
		<dependency>
			<groupId>org.springframework.experimental</groupId>
			<artifactId>spring-graalvm-native</artifactId>
			<version>0.8.1</version>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
			<exclusions>
				<exclusion>
					<groupId>org.junit.vintage</groupId>
					<artifactId>junit-vintage-engine</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
	</dependencies>

	<build>
		<plugins>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
				<configuration>
					<image>
						<builder>paketobuildpacks/builder:tiny</builder>
						<env>
							<BP_BOOT_NATIVE_IMAGE>1</BP_BOOT_NATIVE_IMAGE>
							<BP_BOOT_NATIVE_IMAGE_BUILD_ARGUMENTS>
								-Dspring.native.remove-yaml-support=true
								-Dspring.spel.ignore=true
							</BP_BOOT_NATIVE_IMAGE_BUILD_ARGUMENTS>
						</env>
					</image>
				</configuration>
			</plugin>
		</plugins>
	</build>

	<repositories>
		<repository>
			<id>spring-milestones</id>
			<name>Spring Milestones</name>
			<url>https://repo.spring.io/milestone</url>
		</repository>
		<repository>
			<id>spring-snapshots</id>
			<name>Spring Snapshots</name>
			<url>https://repo.spring.io/snapshot</url>
			<snapshots>
				<enabled>true</enabled>
			</snapshots>
		</repository>
	</repositories>
	<pluginRepositories>
		<pluginRepository>
			<id>spring-milestones</id>
			<name>Spring Milestones</name>
			<url>https://repo.spring.io/milestone</url>
		</pluginRepository>
	</pluginRepositories>
</project>`

`[INFO]     [creator]     [/layers/paketo-buildpacks_spring-boot-native-image/native-image/com.example.restservice.RestServiceApplication:131]        setup:   5,467.48 ms,  2.03 GB
[INFO]     [creator]     Fatal error:java.util.MissingResourceException: Can't find bundle for base name org.apache.tomcat.util.descriptor.tld.LocalStrings, locale en_US
[INFO]     [creator]            at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:1581)
[INFO]     [creator]            at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1396)
`

**Comment by aclement** [comment]
Try this in the `BP_BOOT_NATIVE_IMAGE_BUILD_ARGUMENTS` section as an extra argument:
```
-H:IncludeResourceBundles=org.apache.tomcat.util.descriptor.tld.LocalStrings
```

**Comment by ThanksForAllTheFish** [comment]
The problem is tomcat moved `LocalStrings.properties` up one folder in 9.0.39, and removed `tld` package altogether. I am just playing around with Spring Boot and Graalvm for the first time, and I have no idea how to solve this problem using tomcat 9.0.39 (the new default with Spring Boot 2.4.0-M4: https://docs.spring.io/spring-boot/docs/2.4.0-M4/reference/html/appendix-dependency-versions.html), but downgrading to tomcat 9.0.38 solves this issue.

I guess `spring-boot-native-image` 1.3.2 does not support tomcat 9.0.39, maybe 1.4.0 does, but I am still trying to understand how to change default creators

**Comment by aclement** [comment]
Done some digging. It looks to me like 9.0.39 did remove the package but the `tomcat-resource.json` file they embed that configures the resources for GraalVM to include in native-images was not updated until after that release, so in 9.0.39 it lists something that does not exist and native-image can't cope with that. The way I worked around it was switching to the more optimized programmatic tomcat (better for GraalVM analysis) but I know that isn't an ideal solution:
```
                <dependency>
                        <groupId>org.springframework.boot</groupId>
                        <artifactId>spring-boot-starter-web</artifactId>
                        <exclusions>
                                <exclusion>
                                        <groupId>org.apache.tomcat.embed</groupId>
                                        <artifactId>tomcat-embed-core</artifactId>
                                </exclusion>
                                <exclusion>
                                        <groupId>org.apache.tomcat.embed</groupId>
                                        <artifactId>tomcat-embed-websocket</artifactId>
                                </exclusion>
                        </exclusions>
                </dependency>

                <dependency>
                        <groupId>org.apache.tomcat.experimental</groupId>
                        <artifactId>tomcat-embed-programmatic</artifactId>
                        <version>${tomcat.version}</version>
                </dependency>
```

**Comment by aclement** [comment]
Ok I got creative. In the interim whilst we wait for the release I created a dummy resource bundle that meets the needs of that pattern - it is picked up and gets us beyond the error. For the situation where I could recreate the problem, it worked for me. You will need to pick up a 0.8.2-SNAPSHOT build of spring-graalvm-native to pickup the change in your dependency file (and the build of that including the dummy resource should be through later today).
